docs/proposals/disk-affinity.md

```diff
@@ -0,0 +1,161 @@
+
+## Proposal
+
+Usually for affinity/antiaffinity topology feature is used. But the minimal domain, on which this feature operates, is nodes, not disks.
+So we should use a different approach for this one. Another approach, which is proposed, is to use annotations for pointing out how to use disk affinity/antiaffinity:
+affinity.drives.csi-baremetal.dell.com/type: <pod-bound> - mount PVs for the same pod on the same disk;
+antiaffinity.drives.csi-baremetal.dell.com/type: <pod-label> - specify disk antiaffinity type: currently there is only one option - by pod labels;  
+antiaffinity.drives.csi-baremetal.dell.com/labels: "${list of the labels}" - do not mount PVs on the disks that already have some PVs from other specific pods (specified by labels).
+
+## Implementation
+
+#### Client interface for injecting disk affinity/antiaffinity feature
+
+In order to inject disk affinity/antiaffinity feature client should prepare the corresponding annotations for the application.
+The following is a simple example for nginx StatefulSet:
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+   name: web
+spec:
+   selector:
+      matchLabels:
+         app: nginx
+   serviceName: "nginx"
+   replicas: 3
+   minReadySeconds: 10
+   template:
+      metadata:
+         annotations:
+            affinity.drives.csi-baremetal.dell.com/type: pod-bound                                # here we're specifying to use only one disk for each pod of nginx
+            antiaffinity.drives.csi-baremetal.dell.com/type: pod-label                            # here we're specifying the type of antiaffinity
+            antiaffinity.drives.csi-baremetal.dell.com/labels: ecs-cluster-ss, ecs-cluster-pvg    # here we're specifying not to use the same disks as for ss and pvg 
+         labels:
+            app: nginx
+      spec:
+         terminationGracePeriodSeconds: 10
+         containers:
+            - name: nginx
+              image: k8s.gcr.io/nginx-slim:0.8
+              ports:
+                 - containerPort: 80
+                   name: web
+              volumeMounts:
+                 - name: www-1
+                   mountPath: /usr/share/nginx/html.1
+                 - name: www-2
+                   mountPath: /usr/share/nginx/html.2
+   volumeClaimTemplates:
+      - metadata:
+           name: www-1
+        spec:
+           accessModes: [ "ReadWriteOnce" ]
+           storageClassName: "csi-baremetal-sc-hddlvg"
+           resources:
+              requests:
+                 storage: 1Gi
+      - metadata:
+           name: www-2
+        spec:
+           accessModes: [ "ReadWriteOnce" ]
+           storageClassName: "csi-baremetal-sc-hddlvg"
+           resources:
+              requests:
+                 storage: 1Gi
+
+```
+
+#### Baremetal CSI Driver's improvements:
+
+At Baremetal CSI Driver we should improve Reservation flow (at reservation controller component) for supporting new type of volumes 
```

Comment:
How will this work on a filtering stage of the future standalone CSI scheduler component?

Reply:
Where can I read more about this: some proposals, planning?

Reply:
https://github.com/dell/csi-baremetal/blob/master/docs/proposals/available-capacity-reservation.md

Reply:
Yes, I've already known it. The question about standalone CSI scheduler, not about ACR.

Reply:
Currently in scheduler extender we use `filter` API to perform reservation which is not great, but it doesn't provide `reservation` API for us. But in case of scheduler this API will be exposed. So my question is about filtering stage in scheduler - what steps will be done there with the proposed API? Scheduler feature is not implemented yet, but want to avoid code rewriting for affinity/anti-affinity in the future.

Reply:
Added section about framework based scheduler to proposal